ctrd/container_lock_test.go

```diff
@@ -2,10 +2,50 @@ package ctrd
 
 import (
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
+func Test_containerLock_TrylockWithTimeout(t *testing.T) {
+	l := &containerLock{
+		ids: make(map[string]struct{}),
+	}
+
+	// basically, if the releaseTimeout < the tryLockTimeout,
+	// TrylockWithTimeout will lock successfully. If not, it will fail.
+	runTrylockWithT := func(tryLockTimeout, releaseTimeout time.Duration) bool {
+		id := "c"
+		assert.Equal(t, l.Trylock(id), true)
+		defer l.Unlock(id)
+
+		var (
+			releaseCh = make(chan struct{})
```

Comment:
What is the usage of this variable `releaseCh`?

Reply:
The `releaseCh` variable try to promise that `TrylockWithRetry` will execute before the sleep and unlock function in the main go routine. Otherwise, unlock is done, `TrylockWithRetry` will success immediately.